iele-binary.md

```diff
@@ -370,87 +386,91 @@ After interpreting the strings representing programs as a `WordStack`, it should
-    rule #dasmOpCode(  84 :  _ ) => SLOAD ()
-    rule #dasmOpCode(  85 :  _ ) => SSTORE ()
-    rule #dasmOpCode(  86 :  _ ) => MSIZE ()
-    rule #dasmOpCode(  87 :  _ ) => GAS ()
-    rule #dasmOpCode(  96 :  _ ) => MOVE ()
-    rule #dasmOpCode(  97 : WS ) => #dasmLoad(97, #loadLen(WS), #loadOffset(WS), WS)
-    rule #dasmOpCode(  98 : WS ) => #dasmLoad(98, #loadLen(WS), #loadOffset(WS), WS)
-    rule #dasmOpCode( 100 : W1 : W2 : WS ) => BR(W1 *Int 256 +Int W2)
-    rule #dasmOpCode( 101 : W1 : W2 : WS ) => BRC(W1 *Int 256 +Int W2)
-    rule #dasmOpCode( 102 : W1 : W2 : WS ) => BRLABEL(W1 *Int 256 +Int W2)
-    rule #dasmOpCode( 160 :  _ ) => LOG0 ()
-    rule #dasmOpCode( 161 :  _ ) => LOG1 ()
-    rule #dasmOpCode( 162 :  _ ) => LOG2 ()
-    rule #dasmOpCode( 163 :  _ ) => LOG3 ()
-    rule #dasmOpCode( 164 :  _ ) => LOG4 ()
-    rule #dasmOpCode( 240 : W1 : W2 : W3 : W4 : WS ) => CREATE(W1 *Int 256 +Int W2, W3 *Int 256 +Int W4)
-    rule #dasmOpCode( 241 : W1 : W2 : WS ) => COPYCREATE(W1 *Int 256 +Int W2)
-    rule #dasmOpCode( 242 : W1 : W2 : W3 : W4 : W5 : W6 : WS ) => CALL(W1 *Int 256 +Int W2, W3 *Int 256 +Int W4, W5 *Int 256 +Int W6)
-    rule #dasmOpCode( 243 : W1 : W2 : W3 : W4 : WS ) => CALLDYN(W1 *Int 256 +Int W2, W3 *Int 256 +Int W4)
-    rule #dasmOpCode( 244 : W1 : W2 : W3 : W4 : W5 : W6 : WS ) => STATICCALL(W1 *Int 256 +Int W2, W3 *Int 256 +Int W4, W5 *Int 256 +Int W6)
-    rule #dasmOpCode( 245 : W1 : W2 : W3 : W4 : WS ) => STATICCALLDYN(W1 *Int 256 +Int W2, W3 *Int 256 +Int W4)
-    rule #dasmOpCode( 246 : W1 : W2 : WS ) => RETURN(W1 *Int 256 +Int W2)
-    rule #dasmOpCode( 247 :  _ ) => REVERT()
-    rule #dasmOpCode( 248 : W1 : W2 : W3 : W4 : W5 : W6 : WS ) => LOCALCALL(W1 *Int 256 +Int W2, W3 *Int 256 +Int W4, W5 *Int 256 +Int W6)
-    rule #dasmOpCode( 249 : W1 : W2 : W3 : W4 : WS ) => LOCALCALLDYN(W1 *Int 256 +Int W2, W3 *Int 256 +Int W4)
-    rule #dasmOpCode( 250 : W1 : W2 : WS ) => CALLADDRESS(W1 *Int 256 +Int W2)
-    rule #dasmOpCode( 254 :  _ ) => INVALID ()
-    rule #dasmOpCode( 255 :  _ ) => SELFDESTRUCT ()
-    rule #dasmOpCode( _ ) => encodingError() [owise]
-
-    syntax OpCode ::= #dasmLoad ( Int , Int , Int , WordStack ) [function]
- // ----------------------------------------------------------------------
-    rule #dasmLoad(97, LEN, POS, WS) => LOADPOS(LEN +Int POS, #asUnsigned(WS [ POS .. LEN ]))
-    rule #dasmLoad(98, LEN, POS, WS) => LOADNEG(LEN +Int POS, #asUnsigned(WS [ POS .. LEN ]))
+    syntax OpCode ::= #dasmOpCode (              start: Int, width: Int, bytecode: Bytes ) [function]
+                    | #dasmOpCode ( opcode: Int, start: Int, width: Int, bytecode: Bytes ) [function]
+ // -------------------------------------------------------------------------------------------------
+    rule #dasmOpCode( I, N, BS ) => #dasmOpCode( BS[I], I +Int 1, N -Int 1, BS ) requires N >=Int 1
+    rule #dasmOpCode( _, _, _  ) => encodingError() [owise]
+
+    rule #dasmOpCode(   1 , _, _,  _ ) => ADD ()
```

Comment:
Let's use record syntax for these rules

Reply:
I've addressed this with commit `30a64b9`, let me know if this is what you envisioned.